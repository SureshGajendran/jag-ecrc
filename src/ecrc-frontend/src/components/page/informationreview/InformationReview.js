import React, { useState, useEffect } from "react";
import { Redirect } from "react-router-dom";
import PropTypes from "prop-types";

import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import Table from "../../composite/table/Table";
import { Button } from "../../base/button/Button";
import SideCards from "../../composite/sideCards/SideCards";

export default function InformationReview({
  page: {
    header,
    applicant: {
      firstName,
      middleName,
      lastName,
      birthDate,
      sex,
      street,
      city,
      province,
      postalCode,
      country,
      birthPlace,
      driversLicNo,
      phoneNumber,
      emailAddress,
      jobTitle,
      organizationFacility
    }
  }
}) {
  const [toBack, setToBack] = useState(false);
  const [boxChecked, setBoxChecked] = useState(false);

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const personalInfoElement = [
    { name: "First Name", value: firstName },
    {
      name: "Middle Name",
      value: middleName
    },
    {
      name: "Last Name",
      value: lastName
    },
    {
      name: "City and Country of Birth",
      value: birthPlace
    },
    {
      name: "Birth Date",
      value: birthDate
    },
    {
      name: "Sex",
      value: sex
    }
  ];

  if (driversLicNo) {
    personalInfoElement.push({
      name: "BC Driver's Licence",
      value: driversLicNo
    });
  }

  const personalInfoTable = {
    header: "PERSONAL INFORMATION",
    tableElements: personalInfoElement
  };

  const positionInfoElement = [
    {
      name: "Your Position/Job Title",
      value: jobTitle
    }
  ];

  if (organizationFacility) {
    positionInfoElement.push({
      name: "Organization Facility",
      value: organizationFacility
    });
  }

  const positionInfoTable = {
    header: "POSITION WITH ORGANIZATION",
    tableElements: positionInfoElement
  };

  const addressElement = [
    {
      name: "Street",
      value: street
    },
    {
      name: "City",
      value: city
    },
    {
      name: "Province",
      value: province
    },
    {
      name: "Postal Code",
      value: postalCode
    },
    {
      name: "Country",
      value: country
    },
    {
      name: "Primary Phone Number",
      value: phoneNumber
    },
    {
      name: "Personal Email Address",
      value: emailAddress
    }
  ];

  const addressTable = {
    header: "CONTACT INFORMATION",
    tableElements: addressElement
  };

  const links = [
    {
      name: "I'm an employee or volunteer",
      url: "/tbd"
    },
    {
      name: "Electronic Identity Verification (EIV)",
      url:
        "https://www2.gov.bc.ca/gov/content/safety/crime-prevention/criminal-record-check/electronic-identity-verification-eiv"
    },
    {
      name: "Results and Reconsideration",
      url:
        "https://www2.gov.bc.ca/gov/content/safety/crime-prevention/criminal-record-check/results-and-reconsiderations"
    }
  ];

  const confirmButton = {
    label: "SUBMIT",
    buttonStyle: "btn btn-primary",
    buttonSize: "btn btn-sm",
    type: "submit",
    disabled: !boxChecked
  };

  const confirm = () => {
    // TODO: Check if volunteer, if yes, success, else, cont.
    // TODO: everything required to build payment link, and redirect to said link
  };

  const cancelButton = {
    label: "EDIT APPLICATION",
    buttonStyle: "btn ecrc_accessary_btn",
    buttonSize: "btn btn-sm",
    type: "submit"
  };

  const edit = () => {
    setToBack(true);
  };

  if (toBack) {
    return <Redirect to="/ecrc/applicationform" />;
  }

  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="content col-md-8">
          <h1>Information Review</h1>
          <p>TEXT TO BE CONFIRMED</p>
          <p>
            Please confirm that the information provided s accurate. If it is
            not, please do NOT proceed and contact __________
          </p>
          <Table table={personalInfoTable} />
          <Table table={positionInfoTable} />
          <Table table={addressTable} />
          <div className="declareTitle">DECLARATION</div>
          <section className="declareSection">
            <input
              type="checkbox"
              onClick={() => {
                setBoxChecked(!boxChecked);
              }}
            />
            <span className="declaration-cb">
              I certify that, to the best of my knowledge, the information I
              have provided on my application and will provide as necessary is
              complete, honest and accurate. I understand that a false statement
              or omission of facts herein may lead to a denial of a cannabis
              workers registration. I am also aware that later discovery of an
              omission or misrepresentation may be grounds for any finding of
              suitability to be suspended or revoked.
            </span>
          </section>
          <div className="buttons">
            <Button button={cancelButton} onClick={edit} />
            <Button button={confirmButton} onClick={confirm} />
          </div>
        </div>
        <div className="sidecard">
          <SideCards type={"usefullinks"} sideCardLinks={links} />
          <SideCards type={"contactinformation"} />
          <SideCards type={"collectionnotice"} />
        </div>
      </div>
      <Footer />
    </main>
  );
}

InformationReview.propTypes = {
  page: PropTypes.shape({
    header: PropTypes.shape({
      name: PropTypes.string.isRequired
    }),
    applicant: PropTypes.shape({
      firstName: PropTypes.string.isRequired,
      middleName: PropTypes.string.isRequired,
      lastName: PropTypes.string.isRequired,
      birthDate: PropTypes.string.isRequired,
      sex: PropTypes.string.isRequired,
      street: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired,
      province: PropTypes.string.isRequired,
      postalCode: PropTypes.string.isRequired,
      country: PropTypes.string.isRequired,
      birthPlace: PropTypes.string.isRequired,
      driversLicNo: PropTypes.string,
      phoneNumber: PropTypes.string.isRequired,
      emailAddress: PropTypes.string.isRequired,
      jobTitle: PropTypes.string.isRequired,
      organizationFacility: PropTypes.string
    })
  }).isRequired
};
