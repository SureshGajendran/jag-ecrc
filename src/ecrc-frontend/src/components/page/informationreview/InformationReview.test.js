import React from "react";
import { create } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import InformationReview from "./InformationReview";

describe("InformationReview Component", () => {
  test("Matches the snapshot", () => {
    const header = {
      name: "Criminal Record Check"
    };

    const applicant = {
      legalFirstNm: "Robert",
      legalSecondNm: "Norman",
      legalSurnameNm: "Ross",
      birthPlace: "Daytona Beach, Florida",
      birthDt: "1942-10-29",
      genderTxt: "Male",
      driversLicNo: "1234567",
      phoneNumber: "2501234567",
      emailAddress: "bob.ross@example.com",
      addressLine1: "123 Somewhere",
      cityNm: "Here",
      provinceNm: "British Columbia",
      postalCodeTxt: "V9V 9V9",
      countryNm: "Canada",
      jobTitle: "Painter",
      organizationFacility: ""
    };

    const org = {
      orgApplicantRelationship: "EMPLOYEE",
      orgTicketNumber: "crce",
      defaultScheduleTypeCd: "WBSD",
      defaultCrcScopeLevelCd: "WWCH"
    };

    const setApplicationInfo = () => {};
    const saveApplicant = () => {};
    const saveOrg = () => {};
    const saveApplicationInfo = () => {};

    const page = {
      header,
      applicant,
      org,
      setApplicationInfo,
      saveApplicant,
      saveOrg,
      saveApplicationInfo
    };

    const infoReview = create(
      <MemoryRouter>
        <InformationReview page={page} />
      </MemoryRouter>
    );
    expect(infoReview.toJSON()).toMatchSnapshot();
  });
});
