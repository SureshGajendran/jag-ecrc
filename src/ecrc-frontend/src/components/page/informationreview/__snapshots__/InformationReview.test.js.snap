// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationReview Component Matches the snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="https://www2.gov.bc.ca"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div
          className="navbar-brand"
        >
          Criminal Record Check
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav"
          >
            <a
              className="nav-item nav-link"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              Personal Information
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              First Name
            </td>
            <td>
              Robert
            </td>
          </tr>
          <tr>
            <td>
              Middle Name
            </td>
            <td>
              Norman
            </td>
          </tr>
          <tr>
            <td>
              Last Name
            </td>
            <td>
              Ross
            </td>
          </tr>
          <tr>
            <td>
              Birth Place
            </td>
            <td>
              Daytona Beach, Florida
            </td>
          </tr>
          <tr>
            <td>
              Birth Date
            </td>
            <td>
              1942-10-29
            </td>
          </tr>
          <tr>
            <td>
              Sex
            </td>
            <td>
              Male
            </td>
          </tr>
          <tr>
            <td>
              Primary Phone Number
            </td>
            <td>
              2501234567
            </td>
          </tr>
          <tr>
            <td>
              Personal Email Address
            </td>
            <td>
              bob.ross@example.com
            </td>
          </tr>
          <tr>
            <td>
              BC Driver's Licence Number
            </td>
            <td>
              1234567
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              Position Information
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Job Title
            </td>
            <td>
              Painter
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              Address
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Street
            </td>
            <td>
              123 Somewhere
            </td>
          </tr>
          <tr>
            <td>
              City
            </td>
            <td>
              Here
            </td>
          </tr>
          <tr>
            <td>
              Province
            </td>
            <td>
              British Columbia
            </td>
          </tr>
          <tr>
            <td>
              Postal Code
            </td>
            <td>
              V9V 9V9
            </td>
          </tr>
          <tr>
            <td>
              Country
            </td>
            <td>
              Canada
            </td>
          </tr>
        </tbody>
      </table>
      <p>
        Some kind of checkbox? IIRC
      </p>
      <button
        className="btn btn-primary btn btn-sm"
        onClick={[Function]}
        type="submit"
      >
        Back
      </button>
      <button
        className="btn btn-primary btn btn-sm"
        onClick={[Function]}
        type="submit"
      >
        Confirm
      </button>
    </div>
    <div
      className="sidecard"
    >
      sidecard
    </div>
  </div>
  <footer
    className="footer"
    style={Object {}}
  >
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
